### PR TITLE
Sql-tasks_Ira S-H

### PR DESCRIPTION
**Pull request by irmir**
https://travis-ci.org/github/irmir/db-assignments/builds/727938643

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -78,7 +103,16 @@ async function task_1_4(db) {
...
  */
 async function task_1_5(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        ProductId,
+        ProductName,
+        QuantityPerUnit
+    FROM Products
+    WHERE ProductName LIKE 'A%' OR ProductName LIKE 'B%' OR ProductName LIKE 'C%' OR ProductName LIKE 'D%' OR ProductName LIKE 'E%' OR ProductName LIKE 'F%'
```
Comment (vramaniuk) [inline task/sql-tasks.js]: Please, try to improve this line using RLIKE of REGEXP
https://dev.mysql.com/doc/refman/5.6/en/regexp.html#operator_regexp

File: task/sql-tasks.js [hunk 2]
```diff
@@ -67,7 +83,16 @@ async function task_1_3(db) {
  *
  */
 async function task_1_4(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        Customers.CustomerID AS 'Customer Id',
+        count(OrderID) AS 'Total number of Orders',
+        ROUND( count(OrderID) * 100 / (SELECT count(OrderID) FROM Orders), 5) AS '% of all orders'
+    FROM Customers JOIN Orders ON Customers.CustomerId = Orders.CustomerId
```
Comment (vramaniuk) [inline task/sql-tasks.js]: Actually, we don't need to select from customers and then join orders
Orders table has all needed culumns (CustomerID as well)

File: task/sql-tasks.js [hunk 3]
```diff
@@ -67,7 +83,16 @@ async function task_1_3(db) {
...
 async function task_1_4(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        Customers.CustomerID AS 'Customer Id',
+        count(OrderID) AS 'Total number of Orders',
+        ROUND( count(OrderID) * 100 / (SELECT count(OrderID) FROM Orders), 5) AS '% of all orders'
+    FROM Customers JOIN Orders ON Customers.CustomerId = Orders.CustomerId
+    GROUP BY Customers.CustomerID
+    ORDER BY count(OrderID) * 100 / (SELECT count(OrderID) from Orders) DESC, Customers.CustomerId;
```
Comment (vramaniuk) [inline task/sql-tasks.js]: Please, simplify ORDER BY clause using alias from prevuous step (don't forget escape quotes in sql query in node code)

File: task/sql-tasks.js [hunk 4]
```diff
@@ -247,7 +388,17 @@ async function task_1_18(db) {
...
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT
+        Customers.CustomerID,
+        CompanyName,
+        sum(UnitPrice * Quantity) AS 'TotalOrdersAmount, $'
+    FROM Customers JOIN Orders ON Customers.CustomerID = Orders.CustomerID JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    GROUP BY Customers.CustomerID
+    HAVING sum(UnitPrice * Quantity) > 10000
+    ORDER BY sum(UnitPrice * Quantity) DESC, CustomerID;
```
Comment (vramaniuk) [inline task/sql-tasks.js]: Please, simplify ORDER BY and HAVING clauses using aliases from the prevuous step (don't forget escape quotes in sql query in node code)

File: task/sql-tasks.js [hunk 5]
```diff
@@ -259,17 +410,37 @@ async function task_1_19(db) {
...
  */
 async function task_1_21(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        OrderID,
+        sum(UnitPrice * Quantity) AS 'Maximum Purchase Amount, $'
+    FROM OrderDetails
+    GROUP BY OrderID
+    ORDER BY sum(UnitPrice * Quantity) DESC
```
Comment (vramaniuk) [inline task/sql-tasks.js]: Please, simplify ORDER BY clause using alias from prevuous step (don't forget escape quotes in sql query in node code)

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by irmir** [comment]
Please see my fixes and error https://travis-ci.org/github/irmir/db-assignments/builds/744446814. I don't understand that is why I get this error. I rounded up to 5 in task _1_4 but travis defines differently

**Comment by vramaniuk** [comment]
Ok, I will add you to slack channel "Project_ideas" in advance in order to you have opportunity to join some team and do project.
Please, solve this issue later, compare all the differences with the other's trainees PRs, which Travic builds are succesful.

**Comment by vramaniuk** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Pull request closed without merging**